Server/src/diagramcoordinator.erl

```diff
@@ -2,89 +2,183 @@
 		  Coordinator !  {Did, print_information, ["Node " ++ atom_to_list(To) ++ " received a message from " ++ atom_to_list(From)]}
 	  end,
-	  loop(Coordinator, Did, Pids, Ls, Message_number + 1, [L|[]]);
+	  
+	  loop(Coordinator, Did, Pids, Ls, Message_number + 1, [L|[]], ClassDiagram);
+	  
 	  
 	{previous_message, Coordinator} ->
 	  Coordinator ! {Did, print_information, ["No previous message"]},
-	  loop(Coordinator, Did, Pids, [L|Ls], Message_number, [])
+	  loop(Coordinator, Did, Pids, [L|Ls], Message_number, [], ClassDiagram)
   end;
   
 %This loop runs until the list is empty (when there are no more messages)
-loop(Coordinator, Did, Pids, [L|Ls], Message_number, PrevList) -> 
+loop(Coordinator, Did, Pids, [L|Ls], Message_number, PrevList, ClassDiagram) -> 
   receive
+
+	% Add a class diagram
+  {class_diagram, Classid, Classes, Relations, Coordinator} ->
+		Coordinator ! {class_diagram, Classid, Did, {Classes, Relations}},
+	  Coordinator !  {Did, print_information, ["Linked Class diagram: " ++ atom_to_list(name_classes(Pids, Classes))]},
+  	loop(Coordinator, Did, Pids, [L|Ls], Message_number, PrevList, {Classid, {Classes, Relations}});
+  	
+  	
     {next_message, Coordinator} -> 
       {From, To, Message} = L,
-	  send_message(find_pid(Pids, From), From, To, Message, find_pid(Pids, To), Message_number, Coordinator, Did), 
+	  send_message(find_pid(Pids, From), From, To, Message, find_pid(Pids, To), Message_number, Coordinator, Did, get_class_diagram_id(ClassDiagram)), 
       receive
-	    {message_done, From, To, Message, Message_number} ->
-		  Coordinator ! {message_sent, Did, From, To, Message, Message_number},
-		  %Sends info to the Coordinator that a message has been received by a node. To be printed in the executionlog
-		  Coordinator !  {Did, print_information, ["Node " ++ atom_to_list(To) ++ " received a message from " ++ atom_to_list(From)]}
+	    	{message_done, From, To, Message, Message_number} ->
+		  	Coordinator ! {message_sent, Did, From, To, Message, Message_number},
+		  	%Sends info to the Coordinator that a message has been received by a node. To be printed client-side.
+		  	Coordinator !  {Did, print_information, ["Node " ++ atom_to_list(To) ++ " received a message from " ++ atom_to_list(From)]}
 	  end,
-	  loop(Coordinator, Did, Pids, Ls, Message_number + 1, [L|PrevList]);
+	  loop(Coordinator, Did, Pids, Ls, Message_number + 1, [L|PrevList], ClassDiagram);
+	  
 	  
 	{previous_message, Coordinator} ->
 	  [Prev_H| Prev_T] = PrevList,
 	  List = [L|Ls],
 	  Coordinator ! {previous_confirmation, Did, ["Previous message"]},
-	  loop(Coordinator, Did, Pids, [Prev_H| List], Message_number - 1, Prev_T)
+	  loop(Coordinator, Did, Pids, [Prev_H| List], Message_number - 1, Prev_T, ClassDiagram)
   end.
+
   
-%checks if a class has a process and spawns it if it doesnt exist
+%checks if a class has a process, none if it does not exist.
+find_pid([], _) -> none;
```

Comment:
Have you checked the other places this is used so it doesnt break something else?